src/test/java/org/opendevstack/provision/config/AuthSecurityTestConfig.java

```diff
@@ -0,0 +1,88 @@
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.opendevstack.provision.config;
+
+import static java.util.Map.entry;
+import static org.mockito.Mockito.mock;
+
+import com.atlassian.crowd.service.cache.BasicCache;
+import com.atlassian.crowd.service.cache.CacheImpl;
+import java.util.Map;
+import net.sf.ehcache.CacheManager;
+import org.opendevstack.provision.adapter.IODSAuthnzAdapter;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cache.ehcache.EhCacheManagerFactoryBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Primary;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@Configuration
+public class AuthSecurityTestConfig {
+
+  public static final String TEST_USER_USERNAME = "user";
+  public static final String TEST_ADMIN_USERNAME = "admin";
+  public static final String TEST_ADMIN_EMAIL = "admin@example.com";
+  public static final String TEST_NOT_PERMISSIONED_USER_USERNAME = "not-permissioned-user";
+  public static final String TEST_VALID_CREDENTIAL = "validsecret";
+
+  @Value("${idmanager.group.opendevstack-users}")
+  protected String roleUser;
+
+  @Value("${idmanager.group.opendevstack-administrators}")
+  protected String roleAdmin;
+
+  @Bean
+  public PasswordEncoder passwordEncoder() {
+    return new BCryptPasswordEncoder();
+  }
+
+  @Bean(name = "testUsersAndRoles")
+  public Map<String, String> testUsersAndRoles() {
+    return Map.ofEntries(
+        entry(TEST_USER_USERNAME, roleUser),
+        entry(TEST_ADMIN_USERNAME, roleAdmin),
+        entry(TEST_NOT_PERMISSIONED_USER_USERNAME, "not-opendevstack-user"));
+  }
+
+  @Bean
+  @Primary
+  public IODSAuthnzAdapter iodsAuthnzAdapter() {
+    return mock(IODSAuthnzAdapter.class);
+  }
+
+  @Bean
+  @Primary
+  public BasicCache getCache() {
+    return new CacheImpl(getCacheManager());
+  }
+
+  @Bean
+  @Primary
+  public CacheManager getCacheManager() {
+    return getEhCacheFactory().getObject();
+  }
+
+  @Bean
+  @Primary
+  public EhCacheManagerFactoryBean getEhCacheFactory() {
+    EhCacheManagerFactoryBean factoryBean = new EhCacheManagerFactoryBean();
+    factoryBean.setConfigLocation(new ClassPathResource("crowd-ehcache.xml"));
+    factoryBean.setShared(false);
+    factoryBean.setAcceptExisting(true);
+    return factoryBean;
+  }
```

Comment:
I found the problem that was causing the tests to fail in my dev env. Actually the 3 beans defined above `BasicCache` and `CacheManager` and `EhCacheManagerFactoryBean` are duplicated for the profile `crowd`. The other duplicates are in the class `CrowdSecurityConfiguration', which is not part of the test classes.
After commenting out the beans above my tests run successful in my dev env. 
I propose we remove the beans above to avoid bean duplication for the profile `crowd`. 

As ref: this is the stacktrace that points to the bean duplication:
```
2021-01-20 16:08:04 ERROR o.s.test.context.TestContextManager - Caught exception while allowing TestExecutionListener [org.springframework.test.context.web.ServletTestExecutionListener@775de577] to prepare test instance [
org.opendevstack.provision.services.MailAdapterTest@1587a71]
    java.lang.IllegalStateException: Failed to load ApplicationContext
        at org.springframework.test.context.cache.DefaultCacheAwareContextLoaderDelegate.loadContext(DefaultCacheAwareContextLoaderDelegate.java:132)
        at org.springframework.test.context.support.DefaultTestContext.getApplicationContext(DefaultTestContext.java:123)
        at org.springframework.test.context.web.ServletTestExecutionListener.setUpRequestContextIfNecessary(ServletTestExecutionListener.java:190)
        at org.springframework.test.context.web.ServletTestExecutionListener.prepareTestInstance(ServletTestExecutionListener.java:132)
...
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'getCache' defined in class path resource [org/opendevstack/provision/authentication/crowd/CrowdSecurityConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.atlassian.crowd.service.cache.BasicCache]: Factory method 'getCache' threw exception; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'getCacheManager' defined in class path resource [org/opendevstack/provision/authentication/crowd/CrowdSecurityConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [net.sf.ehcache.CacheManager]: Factory method 'getCacheManager' threw exception; nested exception is org
```